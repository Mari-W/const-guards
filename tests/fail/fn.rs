#![allow(incomplete_features)]
#![feature(generic_const_exprs)]
extern crate const_guards;
use const_guards::guard;

fn main() {
    f::<0>()
}

#[guard(N > 0)]
fn f<const N: usize>() {}
