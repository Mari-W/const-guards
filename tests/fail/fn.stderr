error[E0080]: evaluation of `f::<0_usize>::{constant#0}` failed
  --> tests/fail/fn.rs:10:1
   |
10 | #[guard(<const N: usize> {N > 0})]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | the evaluated program panicked at 'guard evaluated to false', $DIR/tests/fail/fn.rs:10:1
   | inside `_f_guard::<0_usize>` at /home/mari/.rustup/toolchains/nightly-x86_64-unknown-linux-gnu/lib/rustlib/src/rust/library/core/src/panic.rs:57:9
   | inside `f::<0_usize>::{constant#0}` at $DIR/tests/fail/fn.rs:10:1
   |
   = note: this error originates in the macro `$crate::panic::panic_2021` (in Nightly builds, run with -Z macro-backtrace for more info)
